File: 01-array/0110-167.cpp
/*
 * leetcode 167 
 * two sum II - input array is sorted
 *
 * double-pointer: left and right
 *
 */

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int i = 0; 
        int j = numbers.size()-1;
        while (i < j) {
            int s = numbers[i]+numbers[j];
            if (s < target) {
                i++;
            } else if (s > target) {
                j--;
            } else {
                return vector<int>{i+1, j+1};
            }
        }    
        return vector<int>{-1, -1};
    }
};

File: 01-array/0110-167.py
"""
leetcode 167 two sum II

"""

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers)-1
        while True:
            s = numbers[left]+numbers[right]
            if s == target:
                return [left+1, right+1] # index starting from 1 per problem
            if s > target:
                right -= 1
            else:
                left += 1

